lib/Backend/GlobOpt.cpp

```diff
@@ -19069,6 +19107,30 @@ GlobOpt::OptHoistInvariant(
     IR::BailOutKind bailoutKind)
 {
     BasicBlock *landingPad = loop->landingPad;
+
+    IR::Opnd* src1 = instr->GetSrc1();
+    if (src1)
+    {
+        // We are hoisting this instruction possibly past other uses, which might invalidate the last use info. Clear it.
+        OptHoistToLandingPadUpdateValueType(landingPad, instr, src1, src1Val);
```

Comment:
The incorrect value type (if no update) for hoisted instruction could lead incorrect codegen, like missing some type checks which is required for the landingpad. 

---
In reply to: [101557970](https://github.com/Microsoft/ChakraCore/pull/2538#discussion_r101557970) [](ancestors = 101557970)